### PR TITLE
[15.0] Hr applicant Migration

### PR DESCRIPTION
**Pull request by ilakkia-serpentcs**

### CODE REVIEW

File: hr_applicant/models/hr_app_med_occ.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_med_occ.py:4]: Remove extra blank lines

File: hr_applicant/models/hr_app_med_occ.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            "active_id"
+        ):
+            vals.update({"applicant_id": self._context.get("active_id")})
+        return super(ApplicantPreviousOccupation, self).create(vals)
+
+    @api.onchange("from_date", "to_date")
+    def _onchange_date(self):
+        if (
+            self.to_date
+            and fields.Date.from_string(self.to_date)>= fields.Date.today()
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_med_occ.py]: if self.to_date >= fields.Date.today():
Checked everywhere and correct

File: hr_applicant/models/hr_app_rel_edu.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py]: remove line

File: hr_applicant/models/hr_app_rel_edu.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    )
+    active = fields.Boolean(string="Active", default=True)
+    applicant_id = fields.Many2one(
+        "hr.applicant", "Applicant Ref", ondelete="cascade")
+
+    @api.onchange("birthday")
+    def _onchange_birthday(self):
+        if (
+            self.birthday
+            and fields.Date.from_string(self.birthday)>= fields.Date.today()
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py]: same as above

File: hr_applicant/models/hr_app_rel_edu.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+    @api.onchange("relative_type")
+    def _onchange_relative_type(self):
+        if self.relative_type:
+            if self.relative_type in ("Brother", "Father", "Husband", "Son", "Uncle"):
+                self.gender = "Male"
+            elif self.relative_type in ("Mother", "Sister", "Wife", "Aunty"):
+                self.gender = "Female"
+            else:
+                self.gender = ""
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py]: remove else and define self.gender = "" at line 59

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

File: hr_applicant/models/hr_app_med_occ.py (hunk 2 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_med_occ.py]: yes mam checked

File: hr_applicant/models/hr_app_rel_edu.py (hunk 3 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: yes mam removed

File: hr_applicant/models/hr_app_rel_edu.py (hunk 4 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: yes mam changed

File: hr_applicant/models/hr_app_med_occ.py (hunk 1 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_med_occ.py:4]: yes mam removed

Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_med_occ.py:4]: removed extra blank lines and two space line between import and class

File: hr_applicant/models/hr_app_rel_edu.py [hunk 6]
```diff
@@ -137,11 +136,11 @@ def create(self, vals):
 
     @api.onchange("from_date", "to_date")
     def _onchange_date(self):
-        to_date = self.to_date and fields.Date.to_string(self.to_date)
-        if (
-            to_date
-            and fields.Date.from_string(to_date) >= fields.Date.today()
-        ):
+
+        to_date = self.to_date
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py]: not needed, please remove it.

File: hr_applicant/models/hr_app_rel_edu.py [hunk 7]
```diff
@@ -137,11 +136,11 @@ def create(self, vals):
 
     @api.onchange("from_date", "to_date")
     def _onchange_date(self):
-        to_date = self.to_date and fields.Date.to_string(self.to_date)
-        if (
-            to_date
-            and fields.Date.from_string(to_date) >= fields.Date.today()
-        ):
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py:124]: Remove blank lines

File: hr_applicant/models/hr_app_rel_edu.py [hunk 8]
```diff
@@ -137,11 +136,11 @@ def create(self, vals):
...
+        to_date = self.to_date 
+        if (to_date and self.to_date >= fields.Date.today()):
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_rel_edu.py]: if (self.to_date and self.to_date >= fields.Date.today()):

File: hr_applicant/models/hr_app_trav_lang.py [hunk 9]
```diff
@@ -22,6 +22,7 @@ class ApplicantPreviousTravel(models.Model):
 
     @api.model
     def create(self, vals):
+
```
Comment (Trivedi-Vacha-SerpentCS) [inline hr_applicant/models/hr_app_trav_lang.py]: remove blank lines

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

File: hr_applicant/models/hr_app_rel_edu.py (hunk 7 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py:124]: removed lines mam

File: hr_applicant/models/hr_app_rel_edu.py (hunk 8 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: changed to self.to_date

File: hr_applicant/models/hr_app_trav_lang.py (hunk 9 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_trav_lang.py]: removed the space mam

File: hr_applicant/models/hr_app_rel_edu.py (hunk 6 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: removed it mam

File: hr_applicant/models/hr_app_rel_edu.py (hunk 5 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: yes mam, removed

File: hr_applicant/models/hr_app_rel_edu.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+        ],
+        string="Relative Type",
+        required=True,
+    )
+    name = fields.Char(string="Name", size=128, required=True)
+    birthday = fields.Date(string="Date of Birth")
+    place_of_birth = fields.Char(string="Place of Birth", size=128)
+    occupation = fields.Char(string="Occupation", size=128)
+    gender = fields.Selection(
+        [("Male", "Male"), ("Female", "Female")], string="Gender", required=False
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py]:
```suggestion
        [("Male", "Male"), ("Female", "Female")], string="Gender"
```

File: hr_applicant/models/hr_app_rel_edu.py [hunk 11]
```diff
@@ -0,0 +1,151 @@
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py]:
```suggestion
from odoo import _, api, fields, models
from odoo.exceptions import ValidationError
```

File: hr_applicant/models/hr_app_rel_edu.py [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    province = fields.Char("Province")
+
+    @api.onchange("edu_type")
+    def _onchange_edu_type(self):
+        for rec in self:
+            if rec.edu_type == "Local":
+                rec.country_id = False
+            else:
+                rec.province = False
+                rec.state_id = False
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py]:
```suggestion
                rec.province = rec.state_id = False
```

File: hr_applicant/models/hr_app_rel_edu.py [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+            rec.from_date = False
+            rec.to_date = False
+            rec.education_rank = ""
+            rec.school_name = ""
+            rec.grade = ""
+            rec.field = ""
+            rec.edu_type = ""
+            rec.country_id = False
+            rec.state_id = False
+            rec.province = ""
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_app_rel_edu.py]:
```suggestion
            rec.from_date = rec.to_date = rec.country_id = rec.state_id = False
            rec.education_rank = rec.school_name = rec.grade = rec.field = rec.edu_type = rec.province = ""
```

File: hr_applicant/models/hr_app_trav_lang.py [hunk 14]
```diff
@@ -0,0 +1,98 @@
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_app_trav_lang.py]:
```suggestion
from odoo import _, api, fields, models
from odoo.exceptions import ValidationError
```

File: hr_applicant/models/hr_emp_med_occ.py [hunk 15]
```diff
@@ -0,0 +1,102 @@
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_emp_med_occ.py]:
```suggestion
from odoo import _, api, fields, models
from odoo.exceptions import ValidationError
```

File: hr_applicant/models/hr_emp_rel_edu.py [hunk 16]
```diff
@@ -0,0 +1,149 @@
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_emp_rel_edu.py]:
```suggestion
from odoo import _, api, fields, models
from odoo.exceptions import ValidationError
```

File: hr_applicant/models/hr_emp_rel_edu.py [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+        ],
+        string="Relative Type",
+        required=True,
+    )
+    name = fields.Char(string="Name", size=128, required=True)
+    birthday = fields.Date(string="Date of Birth")
+    place_of_birth = fields.Char(string="Place of Birth", size=128)
+    occupation = fields.Char(string="Occupation", size=128)
+    gender = fields.Selection(
+        [("Male", "Male"), ("Female", "Female")], string="Gender", required=False
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_emp_rel_edu.py:33]:
```suggestion
        [("Male", "Male"), ("Female", "Female")], string="Gender"
```

File: hr_applicant/models/hr_recruitment.py [hunk 18]
```diff
@@ -0,0 +1,324 @@
+# See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
```
Comment (nikul-serpentcs) [inline hr_applicant/models/hr_recruitment.py]:
```suggestion
from odoo import _, api, fields, models
from odoo.exceptions import ValidationError
```

File: hr_applicant/wizard/select_training_view.xml [hunk 19]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.ui.view" id="select_training_wizard_form_view">
+        <field name="name">select.training.wizard.form.view</field>
+        <field name="model">select.training</field>
+        <field name="arch" type="xml">
+            <form string="Need Training" version='9.0'>
```
Comment (nikul-serpentcs) [inline hr_applicant/wizard/select_training_view.xml]:
```suggestion
            <form string="Need Training" >
```

File: hr_applicant/views/training_views.xml [hunk 20]
```diff
@@ -0,0 +1,186 @@
...
+        <field name="view_id" ref="view_training_class_tree" />
+    </record>
+
+    <menuitem id="menu_training" name="Training" parent="menu_training_main" action="action_training_class_list"/>
+
+    <record model="ir.ui.view" id="view_list_of_attendees_form">
+        <field name="name">view.list.of.attendees.form</field>
+        <field name="model">list.of.attendees</field>
+        <field name="arch" type="xml">
+            <form string="Training" version="9.0">
```
Comment (nikul-serpentcs) [inline hr_applicant/views/training_views.xml]:
```suggestion
            <form string="Training" >
```

File: hr_applicant/views/training_views.xml [hunk 21]
```diff
@@ -0,0 +1,186 @@
...
+    </record>
+
+    <menuitem id="menu_training_main" name="Training" parent="hr_recruitment.menu_hr_recruitment_root"/>
+    <menuitem id="menu_training_course" name="Training Course" parent="menu_training_main" action="action_training_courses_list"/>
+
+    <record model="ir.ui.view" id="view_training_class_form">
+        <field name="name">view.training.class.form</field>
+        <field name="model">training.class</field>
+        <field name="arch" type="xml">
+            <form string="Training" version="9.0">
```
Comment (nikul-serpentcs) [inline hr_applicant/views/training_views.xml:76]:
```suggestion
            <form string="Training" >
```

File: hr_applicant/views/training_views.xml [hunk 22]
```diff
@@ -0,0 +1,186 @@
...
+                <field name="department"/>
+            </tree>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="view_training_courses_form">
+        <field name="name">view.training.courses.form</field>
+        <field name="model">training.courses</field>
+        <field name="arch" type="xml">
+            <form string="Training Courses" version="9.0">
```
Comment (nikul-serpentcs) [inline hr_applicant/views/training_views.xml:41]:
```suggestion
            <form string="Training Courses" >
```

File: hr_applicant/views/hr_recruitment_views.xml [hunk 23]
```diff
@@ -0,0 +1,325 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.ui.view" id="view_hr_applicant_medical_details_form">
+        <field name="name">view.hr.applicant.medical.details.form</field>
+        <field name="model">hr.applicant.medical.details</field>
+        <field name="arch" type="xml">
+            <form string="Medical Screening" version="9.0">
```
Comment (nikul-serpentcs) [inline hr_applicant/views/hr_recruitment_views.xml]: @ilakkia-serpentcs remove all `version="9.0"`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
nitpicking

File: hr_applicant/models/hr_app_rel_edu.py (hunk 12 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: changes updated

File: hr_applicant/models/hr_app_rel_edu.py (hunk 13 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: code changes updated as per comment

File: hr_applicant/wizard/select_training_view.xml (hunk 19 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/wizard/select_training_view.xml]: removed version= 9.0

File: hr_applicant/models/hr_app_trav_lang.py (hunk 14 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_trav_lang.py]: added  based on comment

File: hr_applicant/views/hr_recruitment_views.xml (hunk 23 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/views/hr_recruitment_views.xml]: removed all  version="9.0" and added report record based on v15

File: hr_applicant/views/training_views.xml (hunk 22 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/views/training_views.xml:41]: removed as per comment

File: hr_applicant/views/training_views.xml (hunk 21 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/views/training_views.xml:76]: updated comment

File: hr_applicant/views/training_views.xml (hunk 20 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/views/training_views.xml]: removed version="9.0"

File: hr_applicant/models/hr_app_rel_edu.py (hunk 11 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: added as per comment

File: hr_applicant/models/hr_app_rel_edu.py (hunk 10 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_app_rel_edu.py]: removed required=False

File: hr_applicant/models/hr_emp_rel_edu.py (hunk 17 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_emp_rel_edu.py:33]: removed required

File: hr_applicant/models/hr_recruitment.py (hunk 18 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_recruitment.py]: added as per comment

File: hr_applicant/models/hr_emp_rel_edu.py (hunk 16 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_emp_rel_edu.py]: added  as per comment

File: hr_applicant/models/hr_emp_med_occ.py (hunk 15 above)
Comment (ilakkia-serpentcs) [reply hr_applicant/models/hr_emp_med_occ.py]: added as per comment

**Pull request merged**